croptalk/model_openai_functions.py

```diff
@@ -101,26 +101,35 @@ def get_model(self) -> Tuple[Chain, AgentTokenBufferMemory]:
             | llm_with_tools
             | OpenAIFunctionsAgentOutputParser()
         )
-        memory_llm = ChatOpenAI(model=self.llm_model_name, temperature=0)
-        memory = AgentTokenBufferMemory(
-            memory_key=self.memory_key,
-            llm=memory_llm,
-            max_token_limit=6000,
-        )
 
-        agent_executor = (
-            AgentExecutor(
+        if no_memory:
+            memory = None
+        else:
+            memory_llm = ChatOpenAI(model=self.llm_model_name, temperature=0)
+            memory = AgentTokenBufferMemory(
+                memory_key=self.memory_key,
+                llm=memory_llm,
+                max_token_limit=6000,
+            )
+
+        agent_executor = AgentExecutor(
                 agent=agent,
                 tools=tools,
                 memory=memory,
                 verbose=True,
                 max_iterations=10,
                 return_intermediate_steps=True,
-            )
+            ).with_config(run_name="AgentExecutor")
+        
+        if return_intermediate_steps:
```

Comment:
I found it a bit confusing that we use `return_intermediate_steps=True` at L121 (i.e. AgentExecutor's param) regardless of whether `return_intermediate_steps` (i.e. get_model's arg) is True or False... maybe we should add a comment to explain why we have both? And why `return_intermediate_steps=return_intermediate_steps` at L121 is not satisfactory?

Reply:
Updated the naming